label/service.go

```diff
@@ -0,0 +1,294 @@
+package label
+
+import (
+	"context"
+	"strings"
+
+	"github.com/influxdata/influxdb/v2"
+	"github.com/influxdata/influxdb/v2/kv"
+)
+
+type Service struct {
+	store      *Store
+	urmCreator UserResourceMappingCreator
+}
+
+type UserResourceMappingCreator interface {
+	CreateUserResourceMappingForOrg(ctx context.Context, tx kv.Tx, orgID influxdb.ID, resID influxdb.ID, resType influxdb.ResourceType) error
+}
+
+func NewService(st *Store, urmCreator UserResourceMappingCreator) influxdb.LabelService {
+	return &Service{
+		store:      st,
+		urmCreator: urmCreator, // todo (al) this can be removed once URMs are removed from the Label service
+	}
+}
+
+// CreateLabel creates a new label.
+func (s *Service) CreateLabel(ctx context.Context, l *influxdb.Label) error {
+	if err := l.Validate(); err != nil {
+		return &influxdb.Error{
+			Code: influxdb.EInvalid,
+			Err:  err,
+		}
+	}
+
+	l.Name = strings.TrimSpace(l.Name)
+
+	err := s.store.Update(ctx, func(tx kv.Tx) error {
+		if err := uniqueLabelName(ctx, tx, l); err != nil {
+			return err
+		}
+
+		if err := s.store.CreateLabel(ctx, tx, l); err != nil {
+			return err
+		}
+
+		return nil
+	})
+
+	if err != nil {
+		return err
+	}
+	err = s.store.Update(ctx, func(tx kv.Tx) error {
+		if err := s.urmCreator.CreateUserResourceMappingForOrg(ctx, tx, l.OrgID, l.ID, influxdb.LabelsResourceType); err != nil {
```

Comment:
Feel free to disregard this. But now that you have expose the transaction as an argument, you can actually do this all one transaction.
The only reason I suggested moving it out of the transaction, was because the new mapping creator service interface wasn't going to expose the transaction. The theory being that it removed any concept of kv from that operation. So we could if we wanted to, swap out the a kv one for a remote service for example. But if we're going to leave in the transactionality, then we can keep it all on one Tx.

Reply:
p.s. please don't let this slow you down, go ahead with it :) Just wanted to clarify the reasoning behind not having the Tx in the method.